File: Sai giresh app/app/src/main/java/com/example/wealthwiseai/ai/RiskAnalyzer.kt
package com.example.wealthwiseai.ai

object RiskAnalyzer {

    data class RiskQuestion(
        val question: String,
        val options: List<String>,
        val points: List<Int> // points corresponding to each option
    )

    val questions = listOf(
        RiskQuestion(
            question = "How stable is your monthly income?",
            options = listOf("Highly Unstable / Variable", "Relatively Stable", "Highly Stable / Guaranteed"),
            points = listOf(1, 2, 3)
        ),
        RiskQuestion(
            question = "How comfortable are you with investment risk?",
            options = listOf(
                "Very uncomfortable: I prefer keeping money safe even if returns are low.",
                "Somewhat comfortable: I can accept mild fluctuations for better returns.",
                "Very comfortable: I can accept temporary losses for high growth potential."
            ),
            points = listOf(1, 2, 3)
        ),
        RiskQuestion(
            question = "How long can you keep money invested?",
            options = listOf("Short term (less than 1 year)", "Medium term (1 to 5 years)", "Long term (more than 5 years)"),
            points = listOf(1, 2, 3)
        ),
        RiskQuestion(
            question = "Do you have emergency savings?",
            options = listOf("None at all", "Some (1 to 3 months of expenses)", "Yes (6+ months of expenses)"),
            points = listOf(1, 2, 3)
        ),
        RiskQuestion(
            question = "What is your investment knowledge level?",
            options = listOf("Beginner: Little to no experience", "Intermediate: Know the basics of mutual funds", "Advanced: Understand stocks and asset allocation"),
            points = listOf(1, 2, 3)
        )
    )

    data class AnalysisResult(
        val score: Int,
        val riskClass: String,
        val explanation: String,
        val learningPath: List<String>
    )

    fun analyzeRisk(answers: List<Int>): AnalysisResult {
        // answers contains the indices selected for each question (0, 1, or 2)
        var totalScore = 0
        for (i in answers.indices) {
            val questionIndex = answers[i]
            val q = questions.getOrNull(i)
            if (q != null && questionIndex in q.options.indices) {
                totalScore += q.points[questionIndex]
            } else {
                totalScore += 1 // fallback
            }
        }

        return when {
            totalScore <= 8 -> AnalysisResult(
                score = totalScore,
                riskClass = "Low Risk",
                explanation = "You prefer capital preservation and security. Stable return profiles and liquid instruments fit your goals best. Avoid volatile assets that cause anxiety.",
                learningPath = listOf("Budgeting basics", "Emergency fund", "Savings habit", "Fixed deposit")
            )
            totalScore <= 12 -> AnalysisResult(
                score = totalScore,
                riskClass = "Medium Risk",
                explanation = "You seek a balanced approach, accepting moderate volatility in exchange for potential growth. A diversified portfolio combining stability with market growth is suitable.",
                learningPath = listOf("Budgeting basics", "Emergency fund", "Mutual funds", "SIP", "Diversification", "Risk management")
            )
            else -> AnalysisResult(
                score = totalScore,
                riskClass = "High Risk",
                explanation = "You are focused on long-term capital appreciation and possess a strong capacity to withstand market fluctuations. You can explore volatile assets to maximize compounding benefits over a long duration.",
                learningPath = listOf("Mutual funds", "SIP", "Stocks basics", "Diversification", "Risk management", "Financial discipline")
            )
        }
    }
}

File: Sai giresh app/frontend/src/lib/ai/riskAnalyzer.ts
export interface RiskAnswer {
  questionIndex: number
  answer: number
}

export const RISK_QUESTIONS = [
  {
    question: "How would you react to a sudden 20% drop in your investments?",
    options: ["Sell everything immediately", "Sell some investments", "Hold and wait", "Buy more at lower price"],
  },
  {
    question: "What is your investment horizon?",
    options: ["Less than 1 year", "1-3 years", "3-7 years", "More than 7 years"],
  },
  {
    question: "Which statement best describes your investment knowledge?",
    options: ["No knowledge", "Basic knowledge", "Moderate knowledge", "Advanced knowledge"],
  },
  {
    question: "What percentage of your monthly income can you comfortably invest?",
    options: ["0-5%", "5-15%", "15-30%", "More than 30%"],
  },
  {
    question: "Your ideal investment return expectation per year is:",
    options: ["5-7% (Fixed deposits)", "8-10% (Debt funds)", "10-12% (Balanced funds)", "12-15%+ (Equity funds)"],
  },
]

export function calculateRiskScore(answers: RiskAnswer[]): { score: number; riskClass: "Low" | "Medium" | "High" } {
  if (answers.length === 0) return { score: 0, riskClass: "Low" }

  const total = answers.reduce((sum, a) => sum + (a.answer + 1), 0)
  const maxScore = RISK_QUESTIONS.length * 4
  const score = Math.round((total / maxScore) * 100)

  let riskClass: "Low" | "Medium" | "High"
  if (score < 40) riskClass = "Low"
  else if (score < 70) riskClass = "Medium"
  else riskClass = "High"

  return { score, riskClass }
}
